spark/src/test/scala/org/apache/spark/sql/delta/IdentityColumnSyncSuite.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta
+
+import org.apache.spark.sql.delta.GeneratedAsIdentityType.GeneratedByDefault
+import org.apache.spark.sql.delta.sources.DeltaSourceUtils
+
+import org.apache.spark.sql.{AnalysisException, Row}
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.types._
+
+
+/**
+ * Identity Column test suite for the SYNC IDENTITY command.
+ */
+trait IdentityColumnSyncSuiteBase
+  extends IdentityColumnTestUtils {
+  private val tblName = "identity_test"
+
+  /**
+   * Create and manage a table with a single identity column "id" generated by default and a single
+   * String "value" column.
+   */
+  private def withSimpleGeneratedByDefaultTable(
+      startsWith: Long, incrementBy: Long)(f: => Unit): Unit = {
+    withTable(tblName) {
+      createTable(
+        tblName,
+        Seq(
+          IdentityColumnSpec(
+            GeneratedByDefault,
+            startsWith = Some(startsWith),
+            incrementBy = Some(incrementBy)),
+          TestColumnSpec(colName = "value", dataType = StringType)
+        )
+      )
+
+      f
+    }
+  }
+
+  test("alter table sync identity delta") {
+    val starts = Seq(-1, 1)
+    val steps = Seq(-3, 3)
+    for (start <- starts; step <- steps) {
+      withSimpleGeneratedByDefaultTable(start, step) {
+        // Test empty table.
+        val oldSchema = DeltaLog.forTable(spark, TableIdentifier(tblName)).snapshot.schema
+        sql(s"ALTER TABLE $tblName ALTER COLUMN id SYNC IDENTITY")
```

Comment:
Can you test the other variants of the SQL syntax? Right now you only test ALTER TABLE ALTER COLUMN <colName> SYNC IDENTITY, but we should validate the SQL syntax for ALTER TABLE CHANGE COLUMN <colName> SYNC IDENTITY, etc.

Reply:
Added test for ALTER TABLE CHANGE COLUMN SYNC IDENTITY